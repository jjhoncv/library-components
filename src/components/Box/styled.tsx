import styled from 'styled-components';

export const Box = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 34px 34px 0;
  width: 100%;
  position: relative;
  z-index: 1;
`;
