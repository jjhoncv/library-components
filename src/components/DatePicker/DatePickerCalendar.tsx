import * as React from 'react';
import * as PropTypes from 'prop-types';
import { useState } from 'react';
import {
  DatePickerCalendarWrapper,
  DatePickerMonth,
  DatePickerPrev,
  DatePickerNext,
  DatePickerTitle,
  DatePickerWeekDays,
  DatePickerWeekDay,
  DatePickerDates,
  DatePickerWeek,
  DatePickerWeekDate
} from './styled';

interface Props {
  date?: number;
  onClick?: Function;
  onChange?: Function;
  isDisabledPreviousDate?: boolean;
}

export const DatePickerCalendar: React.FC<Props> = ({
  date,
  onClick,
  onChange,
  isDisabledPreviousDate
}) => {
  const [ month, setMonth ] = useState(new Date(date).getMonth());
  const [ year, setYear ] = useState(new Date(date).getFullYear());

  const months = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'];
  const monthsName = months[month];
  const weekdays = ['Dom', 'Lun', 'Mar', 'Mie', 'Juv', 'Vie', 'Sab'];
  const totalDays = new Date(year, month + 1, 0).getDate();
  const firstDay = new Date(year, month, 1).getDay();

  const onNext = () => {
    if(month < 11) {
      setMonth(month + 1);
    } else {
      setMonth(0);
      setYear(year + 1);
    }
  }

  const onPrev = () => {
    if(month > 0) {
      setMonth(month - 1);
    } else {
      setMonth(11);
      setYear(year - 1);
    }
  }

  const DatePickerDaysWeek = ({weekdays}) => (
    <DatePickerWeekDays>
      { weekdays.map((weekday, index) => (
        <DatePickerWeekDay key={index}>{weekday}</DatePickerWeekDay>
      )) }
    </DatePickerWeekDays>
  );

  const DatePickerDaysCalendar = ({setDay, onClick}) => {
    let items = [];
    let rows = [];
    let contador = 1;

    for(let count=0; count<42; count++) {
      if(firstDay <= count && count <= totalDays + firstDay - 1) {
        items.push(contador++);
      } else {
        items.push("")
      }
    }

    for(let i=0; i<items.length; i+=7) {
      let sliceIt = items.slice(i, i+7);
      rows.push(sliceIt);
    }

    const handleClick = (e) => {
      let value = e.target.innerText;
      setDay(value);

      if(onClick) {
        onClick(e);
      }
    }

    return (
      <DatePickerDates>
        {rows.map((row, index) => {
          return (
          <DatePickerWeek key={index}>
            {
              row.map((data, i) => {
                const isEmpty = row[i] == "";
                const isLessCurrentDate = new Date(year, month, row[i] + 1).getTime() < new Date().getTime();
                const isSelected = new Date(year, month, row[i]).getTime() == new Date(date).getTime();

                return (
                  isEmpty || (isDisabledPreviousDate && isLessCurrentDate) ? (
                    <DatePickerWeekDate key={i} disabled>{data}</DatePickerWeekDate>) :
                  (isSelected ?
                    (<DatePickerWeekDate key={i} selected onClick={(e) => handleClick(e)}>{data}</DatePickerWeekDate>) :
                    (<DatePickerWeekDate key={i} onClick={(e) => handleClick(e)}>{data}</DatePickerWeekDate>)
                  )
                );
              })
            }
          </DatePickerWeek>
        )})}
      </DatePickerDates>
    );
  }

  const getDay = (day) => {
    onChange(`${formatDate(day)}/${formatDate(month + 1)}/${year}`);
  }

  const formatDate = (date) => {
    return date.toString().length == 1 ? '0' + date : date;
  }

  return (
    <DatePickerCalendarWrapper>
      <DatePickerMonth>
        <DatePickerPrev
          onClick={onPrev}
        >
          ‹
        </DatePickerPrev>
        <DatePickerTitle>{monthsName} {year}</DatePickerTitle>
        <DatePickerNext
          onClick={onNext}>
          ›
        </DatePickerNext>
      </DatePickerMonth>
      <DatePickerDaysWeek weekdays={weekdays}/>
      <DatePickerDaysCalendar setDay={getDay} onClick={onClick}/>
    </DatePickerCalendarWrapper>
  );
}

DatePickerCalendar.propTypes = {
  date: PropTypes.number,
  onClick: PropTypes.func,
  onChange: PropTypes.func,
  isDisabledPreviousDate: PropTypes.bool
}
