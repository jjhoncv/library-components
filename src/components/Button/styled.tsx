import styled, { keyframes } from 'styled-components';
import { COLORS, FONTS } from './../../ui'

interface Props {
  fullWidth?: boolean;
  disabled?: boolean;
  loading?: boolean;
  primary?: boolean;
}

const animationLoading = keyframes`
  to {transform: rotate(360deg);}
`;

export const ButtonLoading = styled.div`
  display: inline-block;
  box-sizing: border-box;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 2px solid ${COLORS.BLACK};
  border-top: 2px solid ${COLORS.LIGHT_GRAY};
  animation: ${animationLoading} .6s linear infinite;

  ${(props: Props) => props.primary && `
    border: 2px solid ${COLORS.WHITE};
    border-top: 2px solid ${COLORS.BLACK};
  `};
`;

export const ButtonStyled = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${(props: Props) => props.fullWidth ? '100%' : 'auto'};
  background-color: ${COLORS.LIGHT_GRAY};
  height: 44px;
  padding: 12px 25px;
  border: none;
  border: 1px solid ${COLORS.OPAQUE_GRAY};
  border-radius: 4px;
  color: ${COLORS.BLACK};
  font-size: 14px;
  font-family: ${FONTS.ROBOTO_REGULAR};
  cursor: pointer;

  &:hover {
    background-color: ${COLORS.HOVER_GRAY};
  }

  ${(props: Props) => props.disabled && `
    pointer-events: none;
    user-select: none;
    opacity: 0.3;
  `};

  ${(props: Props) => props.loading && `
    pointer-events: none;
    user-select: none;
  `};

  ${(props: Props) => props.primary && `
    background:${COLORS.BLACK};
    color: ${COLORS.WHITE};

    &:hover {
      background-color: ${COLORS.HOVER_BLACK};
    }
  `};
`;
