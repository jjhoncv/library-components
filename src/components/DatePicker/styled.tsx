import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

interface Props {
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  disabled?: boolean;
}

interface WeekDateProps {
  disabled?: boolean;
  selected?: boolean;
}

export const DatePickerWrapper = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  width: ${(props: Props) => props.fullWidth ? '100%' : props.width};
  font-family: ${FONTS.ROBOTO_REGULAR};

  ${(props: Props) => props.disabled && `
    pointer-events: none;

    ${DatePickerBox} {
      opacity: 0.3;
    }
  `}
`;

export const DatePickerIcon = styled.div`
  display: flex;
  flex-wrap: wrap;
  position: absolute;
  right: 8px;
  top: 8px;
  cursor: pointer;
  z-index: 1;

  ${(props: Props) => props.error && `
    svg {
      color: black;
    }
  `}
`;

export const DatePickerBox = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  align-items: center;
  width: 100%;
  position: relative;
  z-index: 0;
  color: ${COLORS.BLACK};
  font-size: 14px;
`;

export const DatePickerInput = styled.input`
  background-color: transparent;
  width: 100%;
  height: 40px;
  padding: 12px 30px 12px 12px;
  border: 2px solid ${COLORS.BORDER_GRAY};
  box-sizing: border-box;
  border-radius: 4px;
  cursor: pointer;
  z-index: 2;
  &::placeholder {
    font-family: ${FONTS.ROBOTO_REGULAR};
  }
  &:focus {
    outline: none;
    border: 2px solid ${COLORS.AQUA_BLUE};
  }

  ${(props: Props) => props.error && `
    border: 2px solid ${COLORS.ERROR_RED};
  `}

  &:focus ~ ${DatePickerIcon} {
    svg {
      color: black;
    }
  }
`;

export const DatePickerCalendarWrapper = styled.div`
  position: absolute;
  margin-top: 40px;
  z-index: 9;
  width: 240px;
  max-height: 300px;
  background-color: ${COLORS.WHITE};
  color: ${COLORS.BLACK};
  border: 1px solid ${COLORS.BORDER_GRAY};
  box-shadow: 0px 0px 8px ${COLORS.HOVER_GRAY};
  border-radius: 4px;
  padding: 5px;
`;

export const DatePickerMonth = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: bold;
`;

export const DatePickerPrev = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerNext = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerTitle = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 5 1 0;
  justify-content: center;
`;

export const DatePickerWeekDays = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
`;

export const DatePickerWeekDay = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  font-weight: bold;
`;

export const DatePickerDates = styled.div``;

export const DatePickerWeek = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export const DatePickerWeekDate = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  ${(props: WeekDateProps) => props.disabled && `
    color: ${COLORS.TEXT_GRAY} !important;
    cursor: not-allowed;

    &: hover {
      background: ${COLORS.WHITE} !important;
      color: ${COLORS.TEXT_GRAY} !important;
    }
  `}

  ${(props: WeekDateProps) => props.selected && `
    background-color: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  `}

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;


