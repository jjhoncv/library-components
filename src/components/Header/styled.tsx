import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

export const HeaderWrapper = styled.div`
  width: 100%;
`;
export const HeaderTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_BOLD};
  font-size: 24px;
  line-height: 30px;
  padding-bottom: 7px;
  border-bottom: 1px black dashed;
  color: ${COLORS.TEXT_GRAY};
`;
export const HeaderSubTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_REGULAR};
  font-size: 18px;
  line-height: 23px;
  color: ${COLORS.BLACK};
  padding-top: 9px;
`;

export const HeaderTitleImage = styled(HeaderTitle)`
  display: flex;
  ${({ pos }) => (pos === "start") && `
    justify-content: flex-end;
    flex-direction: row-reverse;
    align-items: flex-end;
  `}

  ${({ pos }) => (pos === "end") && `
    align-items: flex-end;
    justify-content: space-between;
  `}
`;
