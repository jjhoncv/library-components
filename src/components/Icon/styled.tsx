import styled from 'styled-components';
import { COLORS } from './../../ui';

interface Props {
  color?: string,
  size?: string,
  colorHover?: string
}

export const IconStyled = styled.div`
  display: inline-flex;
  flex-wrap: wrap;

  svg {
    color: ${(props: Props) => props.color || `${COLORS.TEXT_GRAY}`}
    ${(props: Props) => `
      width: ${props.size || 'auto'};
      height: ${props.size || 'auto'};
    `}
  }

  &:hover {
    svg {
      color: ${(props: Props) => props.colorHover && `${props.colorHover}`};
    }
  }
`;
