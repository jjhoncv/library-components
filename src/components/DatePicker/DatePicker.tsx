import * as React from 'react';
import * as PropTypes from 'prop-types';
import { useState, useEffect, useRef } from 'react';
import { DatePickerCalendar } from './DatePickerCalendar';
import { Icon } from './../Icon';
import {
  DatePickerWrapper,
  DatePickerBox,
  DatePickerInput,
  DatePickerIcon,
} from './styled';

interface Props {
  date?: string;
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  onChange?: Function;
  isDisabledPreviousDate?: boolean;
  disabled?: boolean;
}

const useDate = (date) => {
  const [ dateFormat, setDateFormat ] = useState(null);
  let dateTime;

  if(date === '') {
    dateTime = new Date().getTime();
  } else {
    const dateSeparated = date.split("/");
    const year = parseInt(dateSeparated[2]);
    const month = parseInt(dateSeparated[1]) - 1;
    const day = parseInt(dateSeparated[0]);

    dateTime = new Date(year, month, day).getTime();
  }

  useEffect(() => {
    setDateFormat(dateTime);
  }, [date]);

  return dateFormat;
}

export const DatePicker:React.FC<Props> = ({
  date = '',
  width,
  fullWidth,
  error,
  isDisabledPreviousDate,
  disabled,
  onChange
}) => {
  const [Â value, setValue ] = useState(date);

  const [ open, setOpen ] = useState(false);

  const ref = useRef(null);

  useEffect(() => {
    document.addEventListener('mousedown', handleOutsideClick);
    return () => {
      document.removeEventListener('mousedown', handleOutsideClick);
    }
  })

  const handleOutsideClick = (e) => {
    const isOutside = !ref.current.contains(e.target);
    isOutside && setOpen(false);
  }

  const dateFormat = useDate(value);

  const handleChange = (date) => {
    setValue(date);
    onChange && onChange(date);
  }

  return (
    <DatePickerWrapper
      ref={ref}
      width={width}
      fullWidth={fullWidth}
      disabled={disabled}
    >
      <DatePickerBox>
        <DatePickerInput
          placeholder="dd/mm/aaaa"
          value={value}
          onClick={() => setOpen(true)}
          onChange={(e) => {setValue(e.target.value); setOpen(false);}}
          error={error}
        />
        <DatePickerIcon error={error}>
          <Icon name="calendar" color="#B6B6B6"/>
        </DatePickerIcon>
      </DatePickerBox>
      { open &&
        (<DatePickerCalendar
          date={dateFormat}
          onChange={handleChange}
          onClick={() => setOpen(false)}
          isDisabledPreviousDate={isDisabledPreviousDate}
        />)
      }
    </DatePickerWrapper>
  );
}

DatePicker.propTypes = {
  date: PropTypes.string,
  width: PropTypes.string,
  fullWidth: PropTypes.bool,
  error: PropTypes.bool,
  isDisabledPreviousDate: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func
}

DatePicker.defaultProps = {
  width: '200px'
}
