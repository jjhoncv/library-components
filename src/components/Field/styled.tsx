import styled from 'styled-components';
import { SelectWrapper } from '../Select/styled';
import { DatePickerWrapper } from '../DatePicker/styled';
import { TimePickerWrapper } from '../TimePicker/styled';
import { TextAreaStyled } from '../TextArea/styled';

import { COLORS, FONTS } from '../../ui';

interface GroupProps {
  keepProps?: boolean;
}

interface Props {
  errorMessage?: string;
  portionWidth?: number;
}

export const FieldStyled = styled.div`
  font-family: ${FONTS.ROBOTO_REGULAR};
  width: ${props => `${props.portionWidth * 100}%`};
  flex: 1 1 100%;


  input {
    width: 100%;
  }
  ${SelectWrapper} {
    width: 100%;
  }
  ${DatePickerWrapper} {
    width: 100%;
  }
  ${TimePickerWrapper} {
    width: 100%;
  }
  ${TextAreaStyled} {
    width: 100%;
  }
  margin-bottom: 14px;

  @media(min-width: 480px) {
    ${(props: Props) => props.portionWidth && `
      flex: 1 1 calc(${props.portionWidth * 100}% - 20px);
    `}
  }
`;

export const FieldGroup = styled.div`
  display: flex;
  flex-wrap: wrap;

  ${(props: GroupProps) => props.keepProps && `
    ${FieldStyled} {
      margin-right: 20px;
      flex: 1 1 calc(50% - 20px);
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  `}

  @media(min-width: 480px) {
    ${FieldStyled} {
      margin-right: 20px;
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  }
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 2px;
  min-height: 19px;
  color: ${COLORS.BLACK};
  font-size: 12px;

  @media(min-width: 480px) {
    font-size: 14px;
  }
`;

export const BoxMessage = styled.span`
  display: block;
  font-size: 12px;
  line-height: 15px;
  margin-top: 4px;
  ${(props: Props) => props.errorMessage && `
    color: ${COLORS.ERROR_RED};
  `}
  height: 15px;
`;
