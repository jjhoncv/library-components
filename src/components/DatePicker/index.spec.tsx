import * as React from 'react';
import { shallow } from 'enzyme';
import { DatePicker } from './DatePicker';
import { DatePickerInput } from './styled';

describe('DatePicker Component', () => {
  describe('DatePicker => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <DatePicker />
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('DatePicker Builder', () => {
    it('DatePicker => Render prop disabled', () => {
      const wrapper = shallow(
        <DatePicker disabled/>
      );
      expect(wrapper.prop('disabled')).toBe(true);
    });
    it('DatePicker => Render prop error', () => {
      const wrapper = shallow(
        <DatePicker error/>
      );
      expect(wrapper.find(DatePickerInput).prop('error')).toBe(true);
    });
    it('DatePicker => Render value input', () => {
      const wrapper = shallow(
        <DatePicker date="13/03/1998"/>
      );
      expect(wrapper.find(DatePickerInput).prop('value')).toBe("13/03/1998");
    });
  });
});
