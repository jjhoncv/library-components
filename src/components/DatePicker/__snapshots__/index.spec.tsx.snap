// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Component DatePicker => render snapshot Render correctly 1`] = `"<div width=\\"200px\\" class=\\"sc-bdVaJa jINyQf\\"><div class=\\"sc-htpNat kADVfa\\"><input placeholder=\\"dd/mm/aaaa\\" value=\\"\\" class=\\"sc-bxivhb bloEHJ\\"/><div class=\\"sc-bwzfXH emNAgD\\"><div color=\\"#B6B6B6\\" class=\\"sc-jTzLTM cOZrOr\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><rect x=\\"3\\" y=\\"4\\" width=\\"18\\" height=\\"18\\" rx=\\"2\\" ry=\\"2\\"></rect><line x1=\\"16\\" y1=\\"2\\" x2=\\"16\\" y2=\\"6\\"></line><line x1=\\"8\\" y1=\\"2\\" x2=\\"8\\" y2=\\"6\\"></line><line x1=\\"3\\" y1=\\"10\\" x2=\\"21\\" y2=\\"10\\"></line></svg></div></div></div></div>"`;
